@xen-orchestra/proxy/src/app/mixins/reverseProxy.mjs

```diff
@@ -0,0 +1,117 @@
+import { urlToHttpOptions } from 'url' // node  14.18.0, 15.7.0 is it ok ?
+import proxy from 'http2-proxy'
+
+function removeSlash(str) {
+  return str.replace(/^\/|\/$/g, '')
+}
+
+function mergeUrl(relative, base) {
+  const res = new URL(base)
+  const relativeUrl = new URL(relative, base)
+  res.pathname = relativeUrl.pathname
+  relativeUrl.searchParams.forEach((value, name) => {
+    // we do not allow to modify params already specified by config
+    if (!res.searchParams.has(name)) {
+      res.searchParams.append(name, value)
+    }
+  })
+  res.hash = relativeUrl.hash.length > 0 ? relativeUrl.hash : res.hash
+  return res
+}
+export default class ReverseProxy {
+  constructor(app, { httpServer }) {
+    this._app = app
+    httpServer.on('request', (req, res) => this.proxy(req, res))
+    httpServer.on('upgrade', (req, socket, head) => this.upgrade(req, socket, head))
+  }
+
+  localToBackendUrl(basePath, target, localPath) {
+    let localPathWithoutBase = removeSlash(localPath).substring(basePath.length)
+    localPathWithoutBase = './' + removeSlash(localPathWithoutBase)
+    const url = mergeUrl(localPathWithoutBase, target)
+    return url
+  }
```

Comment:
Does not reference `this`, should be a separate function.

Reply:
a separate private function won't be exported and won't be testable easily

Reply:
The ideal would be to move this function in a separate module and test it independently.

An easier solution would be to export this utility directly from this module without making it a method, at least it won't be exposed to the rest of the application (via `app.reverseProxy.localToBackendUrl`).